src/logger/loggerApi.c

```diff
@@ -1352,6 +1347,133 @@ int api_setCanConfig(struct Serial *serial, const jsmntok_t *json)
+{
+    /* Validate parameters */
+    uint8_t bit_mode = mapping->bit_mode;
+    jsmn_exists_set_val_bool(json_mapping, "bm", &bit_mode);
+
+    uint8_t offset = mapping->offset;
+    jsmn_exists_set_val_uchar(json_mapping, "offset", &offset, NULL);
+
+    uint8_t length = mapping->length;
+    jsmn_exists_set_val_uchar(json_mapping, "len", &length, NULL);
+
+    /* check for invalid mapping offsets and length */
+    if (offset > MAX_CAN_MAPPING_OFFSET_BYTES * (bit_mode? 8 : 1))
+            return API_ERROR_PARAMETER;
+
+    if (length > MAX_CAN_MAPPING_OFFSET_BYTES * (bit_mode? 8 : 1))
+            return API_ERROR_PARAMETER;
+
+    /* we made it this far, let's update the mapping */
+    mapping->bit_mode = bit_mode;
+    mapping->offset = offset;
+    mapping->length = length;
+    uint8_t can_bus_channel = 0;
+    jsmn_exists_set_val_uchar(json_mapping, "bus", &can_bus_channel, filter_can_channel);
+    mapping->can_channel = can_bus_channel;
+
+    jsmn_exists_set_val_int(json_mapping, "id", &mapping->can_id);
+    jsmn_exists_set_val_int(json_mapping, "id_mask", &mapping->can_mask);
+    jsmn_exists_set_val_bool(json_mapping, "bigEndian", &mapping->big_endian);
+    jsmn_exists_set_val_float(json_mapping, "mult", &mapping->multiplier);
+    jsmn_exists_set_val_float(json_mapping, "div", &mapping->divider);
+    jsmn_exists_set_val_float(json_mapping, "add", &mapping->adder);
+    jsmn_exists_set_val_uchar(json_mapping, "filtId", &mapping->conversion_filter_id, NULL);
+    return API_SUCCESS;
+}
+
+int api_set_can_channel_config(struct Serial *serial, const jsmntok_t *json)
+{
+    CANChannelConfig * can_channel_cfg = &(getWorkingLoggerConfig()->can_channel_cfg);
+
+    /* flag to indicate if this channel is the last in a series */
+    bool last = false;
+    jsmn_exists_set_val_bool(json, "last", &last);
+
+    /* optional starting index. start at beginning by default */
+    unsigned index = 0;
+    jsmn_exists_set_val_int(json, "index", &index);
+
+    /* we can only start updating up to the item right after the last */
+    if (index >= CONFIG_CAN_MAPPINGS || index > can_channel_cfg->enabled_mappings)
+            return API_ERROR_PARAMETER;
+
+    /* find the beginning of the channels json array */
+    const jsmntok_t *chans_tok = jsmn_find_node(json, "chans");
+    chans_tok = jsmn_find_node_type(chans_tok, JSMN_ARRAY);
+
+    if (chans_tok) {
+        int channel_max = chans_tok->size;
+        if (channel_max > MAX_CAN_MESSAGE_CHANNELS) {
+            return API_ERROR_PARAMETER;
+        }
+
+        channel_max += index;
+        if (channel_max > CONFIG_CAN_MAPPINGS) {
+            return API_ERROR_PARAMETER;
+        }
+
+        for (chans_tok++; index < channel_max; index++) {
+            CANChannel *pid_cfg = can_channel_cfg->can_channels + index;
+            ChannelConfig *chCfg = &(pid_cfg->channel_cfg);
+
+            int result = _set_can_mapping(chans_tok, &(pid_cfg->mapping));
```

Comment:
since it's a large chunk of memory it's not feasible to create a local copy.  Might need to do a filtering technique where we rail invalid values.  Will think about this.